packages/core/test/unit/segments/segment.test.js

```diff
@@ -262,6 +262,57 @@ describe('Segment', function() {
     });
   });
 
+  describe('#addSubsegmentWithoutSampling', function (){
+    it('should have isSampled flag set to false', function(){
+      var segment = new Segment('parent');
+      var child = new Subsegment('child');
+      segment.addSubsegmentWithoutSampling(child);
+
+      assert.equal(child.isSampled, false);
+    })
+
+    it('should have isSampled flag set to false for new subsegment', function(){
+      var segment = new Segment('parent');
+      var child = segment.addNewSubsegmentWithoutSampling('child');
+
+      assert.equal(child.isSampled, false);
+    })
+
+    it('should not contain the child subsegment if not sampled', function(){
+      var segment = new Segment('parent');
+      var child = new Subsegment('child');
+      segment.addSubsegmentWithoutSampling(child);
+
+      assert.notEqual(segment.subsegments[0], child);
+    })
+
+
+    it('should not sample subsegment or subsegment of subsegment', function(){
+      var segment = new Segment('parent');
+      var child = new Subsegment('child');
+      var child2 = new Subsegment('child-2');
+      segment.addSubsegmentWithoutSampling(child);
+      child.addSubsegmentWithoutSampling(child2)
+
+      assert.equal(child.isSampled, false);
+      assert.equal(child2.isSampled, false);
+    })
+
+    it('should not sample subsegment or subsegment of subsegment - mix', function(){
+      var segment = new Segment('parent');
```

Comment:
Us needing to create the segment here signals that we re not running through the lambda specific code.

When the code thinks it is in lambda it will create a facade segment for us. The facade segment's trace data is filled out by what is in this variable. You can see an example in aws_lambda.test.js

process.env._X_AMZN_TRACE_ID
packages/core/test/unit/env/aws_lambda.test.js

I believe you need to call "init" here with that variable set.
packages/core/lib/env/aws_lambda.js

Reply:
Thanks for pointing that out! I've incorporated that change into the updated unit tests

Reply:
To answer the question above, yes, this was tested in Lambda using the sample sample JSON data used in the Java oversampling mitigation PR. 

For trace header with sampled=1:
<img width="1405" alt="image (3)" src="https://user-images.githubusercontent.com/60774943/196601148-cae368d3-6c86-4d9e-943f-28324e86421c.png">

For trace header with sampled=0:
<img width="1412" alt="image (4)" src="https://user-images.githubusercontent.com/60774943/196601206-117ae11a-802d-43cd-92a2-f57a9d446fde.png">